src/main/kotlin/com/bantolomeus/repository/BookRepository.kt

```diff
@@ -12,37 +12,41 @@ const val BOOK_FILE = "books.json"
 
 @Repository
 class BookRepository(private val bookFile: String = BOOK_FILE) {
-    private val objectMapper = ObjectMapper()
+    private val objectMapper = jacksonObjectMapper()
 
-    fun saveBooks(books: BooksFileDTO): BooksFileDTO {
+    fun saveBooks(books: List<BookDTO>): List<BookDTO> {
         objectMapper.writeValue(File(bookFile), books)
         return books
     }
 
-    fun saveBookIfItNotExists(bookDTO: BookDTO): BookDTO {
-        return if (getBookByName(bookDTO.name).name != bookDTO.name) {
+    fun saveBookIfItNotExists(newVersion: BookDTO): BookDTO {
```

Comment:
For me the name newVersion makes no sense. I want to save a book that contains the data from BookDTO if it does not exist. 
newVersion implies for me that I already have another version, which can happen but is not always the case, but the variable name implies that it is always the case.

Reply:
If I read the variable name bookDTO I know it is a book. newVersion does not tell me what kind of new version, ofc I can look up the type but that is one more step for me. You could also say that I can remember the type in this short method, but if I can not I have more work.

Reply:
And I would not distinguish between new and old or existing data, because that complicates the understanding. I have a book, wanna save it if it does not exist, I look if it exists, if it does not exist i save the book, else I do not save it. The return value should differ between saved book and not saved book.
If I have a new version I always think there exists already another version and I will never save the newVersion.

Reply:
I knew this would come up, but I wanted to start the discussion about the two issues you were mentioning
1) Context
the method name `saveBookIfItNotExists` should give enough context about what is happening in the method and that it's not about versions of an apple or an old version of a box that is being saved if they not exist (*save* by the way is also misleading. Isn't this method *creating* objects?). Anyway, I remember this blog posts about a good method length of 8 lines. Everything that happens within a method should fit in your brain at the same time. So if a method is short enough, then it should be not hard to remember that both the words new and old refer to a book because it's all written down and gives good method names more importance which is also a good thing.

Although I agree, that it's not about a new version but rather about `new` vs. `existing` or `newBook` vs. `existingBook` or maybe `book` vs. `existingVersion`?

2) Meaning vs. Data Type
You say you don't wanna distinguish. But the code was already distinguishing. Not between old and new but between two types of objects which is even harder to understand because there's no context in the name `xyzDTO` (what's the difference between `book` vs. `bookDTO`, where actually both are type bookDTO ?). To me it feels very natural to give names based on the meaning of a variable as opposed to names based on data types. If you have problems doing that and different variables are actually doing the same thing, then maybe you don't understand the method or there's something wrong with the code.

Reply:
newBook and existingBook